packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/lpa-questionnaire-submission/repo.js

```diff
@@ -0,0 +1,102 @@
+const { createPrismaClient } = require('#db-client');
+const { PrismaClientKnownRequestError } = require('@prisma/client/runtime/library');
+
+/**
+ * @typedef {import('@prisma/client').Prisma.LPAQuestionnaireSubmissionGetPayload<
+ * {
+ * 	include: {
+ *		AppealCase: {
+ *			select: {
+ *				LPACode: true
+ *			}
+ *		}
+ * 	}
+ * }
+ * >} LPAQuestionnaireSubmission
+ */
+
+class LPAQuestionnaireSubmissionRepository {
+	dbClient;
+
+	constructor() {
+		this.dbClient = createPrismaClient();
+	}
+
+	/**
+	 * Get questionnaire for given appeal
+	 *
+	 * @param {string} caseReference
+	 * @returns {Promise<LPAQuestionnaireSubmission|null>}
+	 */
+	async getLPAQuestionnaireByAppealRef(caseReference) {
+		try {
+			return await this.dbClient.lPAQuestionnaireSubmission.findUnique({
+				where: {
+					appealCaseReference: caseReference
+				},
+				include: {
+					AppealCase: {
+						select: {
+							LPACode: true
+						}
+					},
+					SubmissionDocumentUpload: true,
+					SubmissionNeighbourAddress: true
+				}
+			});
+		} catch (e) {
+			if (e instanceof PrismaClientKnownRequestError) {
+				if (e.code === 'P2023') {
+					// probably an invalid ID/GUID
+					return null;
+				}
+			}
+			throw e;
+		}
+	}
+
+	/**
+	 * Create questionnaire for given appeal
+	 *
+	 * @param {string} caseReference
+	 * @returns {Promise<LPAQuestionnaireSubmission>}
+	 */
+	async createQuestionnaire(caseReference) {
+		return await this.dbClient.lPAQuestionnaireSubmission.create({
+			data: {
+				appealCaseReference: caseReference
+			},
+			include: {
+				AppealCase: {
+					select: {
+						LPACode: true
+					}
+				}
+			}
+		});
+	}
+
+	/**
+	 *
+	 * @param {*} caseReference
+	 * @param {*} data
+	 * @returns {Promise<LPAQuestionnaireSubmission>}
+	 */
+	async patchLPAQuestionnaireByAppealId(caseReference, data) {
+		return await this.dbClient.lPAQuestionnaireSubmission.update({
+			where: {
+				appealCaseReference: caseReference
+			},
+			data,
```

Comment:
Looks like this data just gets passed in from the service, is there anywhere that checks this data looks like what we expect or will Prisma complain if that happens?

Reply:
Ah, good spot. yes Prisma does complain - but he API then returned 500 which isn't right. I've handled that 400 case now, and added PATCH to the spec.

[See https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3404/commits/4e8fa0b65ec136edab9146982e2baba56120214d](https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3404/commits/8dbf0a55eb3f90df2fe9bbcabc4e77b3a2f828f3)